pkg/apis/core/v1beta1/defaults_seed.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/utils/pointer"
+)
+
+// SetDefaults_Seed sets default values for Seed objects.
+func SetDefaults_Seed(obj *Seed) {
+	if obj.Spec.Settings == nil {
+		obj.Spec.Settings = &SeedSettings{}
+	}
+
+	var defaultExcessCapacityReservationConfigs = []SeedSettingExcessCapacityReservationConfig{
+		// This roughly corresponds to a single, moderately large control-plane.
+		{Resources: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("2"), corev1.ResourceMemory: resource.MustParse("6Gi")}},
+	}
+
+	if obj.Spec.Settings.ExcessCapacityReservation == nil {
+		obj.Spec.Settings.ExcessCapacityReservation = &SeedSettingExcessCapacityReservation{
+			Configs: defaultExcessCapacityReservationConfigs,
+		}
+	}
+
+	if pointer.BoolDeref(obj.Spec.Settings.ExcessCapacityReservation.Enabled, true) && len(obj.Spec.Settings.ExcessCapacityReservation.Configs) == 0 {
+		obj.Spec.Settings.ExcessCapacityReservation.Configs = defaultExcessCapacityReservationConfigs
+	}
+
+	if obj.Spec.Settings.Scheduling == nil {
+		obj.Spec.Settings.Scheduling = &SeedSettingScheduling{Visible: true}
+	}
+
+	if obj.Spec.Settings.VerticalPodAutoscaler == nil {
+		obj.Spec.Settings.VerticalPodAutoscaler = &SeedSettingVerticalPodAutoscaler{Enabled: true}
+	}
+
+	if obj.Spec.Settings.DependencyWatchdog == nil {
+		obj.Spec.Settings.DependencyWatchdog = &SeedSettingDependencyWatchdog{}
+	}
+
+	if obj.Spec.Settings.TopologyAwareRouting == nil {
+		obj.Spec.Settings.TopologyAwareRouting = &SeedSettingTopologyAwareRouting{Enabled: false}
+	}
```

Comment:
This func is called `SetDefaults_Seed` but it is mainly defaulting SeedSettings? Shouldn't the SeedSettings defaulting be extracted to `SetDefaults_SeedSettings`?